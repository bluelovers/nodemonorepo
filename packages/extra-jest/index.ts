import * as snap from './snap'
export = {snap}
