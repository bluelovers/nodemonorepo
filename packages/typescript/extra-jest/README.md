# extra-jest

Some extra functions to work with jest testing framework

## Requirements

* Node.js ≥ 8.9.0

## APIs

**NOTE:**
  * `xjest` is an alias of module `extra-jest`
  * `jsYaml` is an alias of module `js-yaml`
  * `fsTreeUtils` is an alias of module `fs-tree-utils`

### `xjest.snap.mkfn`

```typescript
type DumpFunction<X> = (x: X, options?: jsYaml.DumpOptions) => string
type TestFunction<X> = (x: X, options?: jsYaml.DumpOptions) => () => void

declare function mkfn<X> (fn: DumpFunction<X>, defaultOptions?: jsYaml.DumpOptions): TestFunction<X>
```

This function is used to created `xjest.snap.*` functions below.

### `xjest.snap.*`

```typescript
declare function unsafe (x: any, options?: jsYaml.DumpOptions): () => void
declare function safe (x: any, options?: jsYaml.DumpOptions): () => void
declare function pureSafe (x: any, options?: jsYaml.DumpOptions): () => void
declare function pureUnsafe (x: any, options?: jsYaml.DumpOptions): () => void
declare function noRefs (x: any, options?: jsYaml.DumpOptions): () => void
```

Create jest snapshots but in YAML format.

**Features:**
  * `xjest.snap.unsafe` dumps JavaScript objects before writing to snapshot.
  * `xjest.snap.safe` safely dumps JavaScript objects before writing to snapshot.
  * `xjest.snap.pureUnsafe` is `xjest.snap.unsafe` without default options.
  * `xjest.snap.pureSafe` is `xjest.snap.safe` without default options.
  * `xjest.snap.noRefs` is `xjest.snap.unsafe` with self references.

**Default Options:**

```typescript
namespace defaultOptions {
  export const skipInvalid = true
  export const sortKeys = true
  export const noRefs = false
  export const noCompatMode = true
}
```

### `xjest.setupTeardown.base.createFactory`

```typescript
type PromiseFunc<X, Y> = (x: X) => Promise<Y>
type SyncFunc<X, Y> = (x: X) => Y
type SetupFunc<Y> = () => Promise<Y>
type TeardownFunc<X> = (x: X) => Promise<void>
type CalledFunc<X, Y> = PromiseFunc<X, Y>
type SyncCalledFunc<X, Y> = SyncFunc<X, Y>
type Tester = () => Promise<void>
type AsyncTesterFactory<SM, MT> = (fn: CalledFunc<SM, MT>) => Tester
type SyncTesterFactory<SM, MT> = (fn: SyncCalledFunc<SM, MT>) => Tester

interface Config<SY, TX> {
  readonly setup: SetupFunc<SY>
  readonly teardown: TeardownFunc<TX>
}

type TesterFactory<SM, MT> = AsyncTesterFactory<SM, MT> & {
  forAsync: AsyncTesterFactory<SM, MT>,
  forSync: SyncTesterFactory<SM, MT>
}

declare function createFactory<SM, MT> (config: Config<SM, MT>): TesterFactory<SM, MT>
```

Creates a `TesterFactory<SM, MT>` from a `config: Config<SM, MT>`.

**Parameters:**
  * `config.setup`: A function that runs before main test.
  * `config.teardown`: A function that runs after main test.

**Returns:**
  * A `TesterFactory<SM, MT>` is both a namespace and a function, it has the same effect as `TesterFactory<SM, MT>::forAsync`.
  * A `TesterFactory::forAsync` is a function, it creates a `Tester` from `fn: CalledFunc<SM, MT>`.
  * A `TesterFactory::forSync` is a function, it creates a `Tester` from `fn: SyncCalledFunc<SM, MT>`.
  * A `Tester` can be passed into jest `test` functions as a test function (second argument).

**Aliases:**
  * `xjest.setupTeardown.createFactory`
  * `xjest.setupTeardown.default`

### `xjest.setupTeardown.virtualEnvironment.createFactory`

```typescript
interface Info {
  readonly tree: Tree.Write.Node
  readonly container: string
  readonly previousWorkingDirectory: string
}

interface Factory {
  readonly info: Info
  readonly apply: base.TesterFactory<Info, void>
}

declare function createFactory (tree: fsTreeUtils.Tree.Write.Node, container?: string): Factory
```

**Parameters:**
  * `tree`: Specifies directory structure of target directory.
  * `container` (optional): Intended target directory, default to random name in `os.tmpdir()` folder.

**Returns:**
  * `Factory::info`: Contains some information that might be useful (see `interface Info`).
  * `Factory::apply`: Takes an async function and returns a `Tester`.

### `xjest.snapSpawn.snap`

```typescript
type SpawnFunc = (argv: string[], options: SpawnSyncOptions) => SpawnSyncReturns<string | Buffer>
declare function snap (fn: SpawnFunc, argv?: string[], options?: SpawnSyncOptions): () => void
```

Create a function that creates a snapshot from result of process execution.

**Parameters:**
  * `fn`: A function that calls `spawnSync`.
  * `argv` (optional): Additional arguments.
  * `options` (optional): Options to pass to `spawnSync`.
  * `snap` (optional): Snap function to use.

**Returns:**
  * A function that makes snapshots from process execution result.

**Aliases:**
  * `xjest.snapSpawn.snap.default`

### `xjest.snapSpawn.snap.withCommand`

Like `xjest.snapSpawn.snap` above but with `command: string` as the first parameter instead of `fn: SpawnFunc`.

## License

[MIT](https://git.io/vhaEz) © [Hoàng Văn Khải](https://github.com/KSXGitHub)
