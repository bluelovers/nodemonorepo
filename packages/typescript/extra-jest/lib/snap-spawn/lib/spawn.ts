import { SpawnSyncOptions, SpawnSyncReturns, spawnSync } from 'child_process'
import ramda from 'ramda'

export type IOData = Buffer | string
export type OptionalIOData = IOData | null | undefined

const fmtNull = <X>(x: X) =>
  x == null ? null : x

const fmtStdIO = (buf: OptionalIOData): string | null => {
  if (buf == null) return null
  const str = String(buf)
  return str.trim() ? `\n${str}\n` : '((EMPTY))'
}

export type Argv = string[]

export interface SpawnReturns {
  readonly status: number | null
  readonly signal: string | null
  readonly error: Error | null
  readonly stdout: string | null
  readonly stderr: string | null
}

export type SpawnFunc = (argv: Argv, options: SpawnSyncOptions) => SpawnSyncReturns<IOData>

export const createSpawnFunc = (command: string): SpawnFunc => ramda.partial(spawnSync, [command])

export function spawn (
  fn: SpawnFunc,
  argv: Argv = [],
  options: SpawnSyncOptions = {}
): SpawnReturns {
  const { status, signal, error, stdout, stderr } = fn(
    argv,
    options
  )

  return {
    status: fmtNull(status),
    signal: fmtNull(signal),
    error: fmtNull(error),
    stdout: fmtStdIO(stdout),
    stderr: fmtStdIO(stderr)
  }
}

export namespace spawn {
  export const withCommand = (
    command: string,
    argv?: Argv,
    options?: SpawnSyncOptions
  ) => spawn(createSpawnFunc(command), argv, options)
}

export default spawn
