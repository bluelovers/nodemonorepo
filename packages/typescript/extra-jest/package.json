{
  "name": "extra-jest",
  "version": "0.0.17",
  "description": "Some extra functions to work with jest testing framework",
  "main": "index.js",
  "author": "Hoàng Văn Khải <hvksmr1996@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/ksxnodemodules/nodemonorepo#readme",
  "bugs": {
    "url": "https://github.com/ksxnodemodules/nodemonorepo/issues"
  },
  "keywords": [
    "jest"
  ],
  "dependencies": {
    "ramda": "^0.25.0",
    "fs-extra": "^6.0.1",
    "tslib": "^1.9.2",
    "fs-tree-utils": "^0.0.14",
    "unique-temp-path": "^0.0.9",
    "@types/ramda": "^0.25.32",
    "@types/fs-extra": "^5.0.2",
    "@types/node": "^10.3.0"
  },
  "peerDependencies": {
    "jest": "^23.1.0"
  },
  "engines": {
    "node": ">= 8.9.0"
  }
}
