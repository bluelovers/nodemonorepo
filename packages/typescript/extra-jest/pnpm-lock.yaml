dependencies:
  '@types/fs-extra': 8.0.0
  '@types/js-yaml': 3.12.1
  '@types/node': 12.7.1
  '@types/ramda': 0.26.19
  fs-extra: 8.1.0
  fs-tree-utils: 'link:../fs-tree-utils'
  js-yaml: 3.13.1
  ramda: 0.26.1
  tslib: 1.10.0
  unique-temp-path: 'link:../unique-temp-path'
lockfileVersion: 5.1
packages:
  /@types/fs-extra/8.0.0:
    dependencies:
      '@types/node': 12.7.1
    dev: false
    resolution:
      integrity: sha512-bCtL5v9zdbQW86yexOlXWTEGvLNqWxMFyi7gQA7Gcthbezr2cPSOb8SkESVKA937QD5cIwOFLDFt0MQoXOEr9Q==
  /@types/js-yaml/3.12.1:
    dev: false
    resolution:
      integrity: sha512-SGGAhXLHDx+PK4YLNcNGa6goPf9XRWQNAUUbffkwVGGXIxmDKWyGGL4inzq2sPmExu431Ekb9aEMn9BkPqEYFA==
  /@types/node/12.7.1:
    dev: false
    resolution:
      integrity: sha512-aK9jxMypeSrhiYofWWBf/T7O+KwaiAHzM4sveCdWPn71lzUSMimRnKzhXDKfKwV1kWoBo2P1aGgaIYGLf9/ljw==
  /@types/ramda/0.26.19:
    dependencies:
      ts-toolbelt: 3.6.1
    dev: false
    resolution:
      integrity: sha512-lJuMZDXqZ8W2hl0Gz+wEPUhXcf/tZRuaO5vrxFD3Dv2XMqgTZVw+bvfS6iQAsio9XQV+uhkggtT4eLhDmUpI5g==
  /argparse/1.0.10:
    dependencies:
      sprintf-js: 1.0.3
    dev: false
    resolution:
      integrity: sha512-o5Roy6tNG4SL/FOkCAN6RzjiakZS25RLYFrcMttJqbdd8BWrnA+fGz57iN5Pb06pvBGvl5gQ0B48dJlslXvoTg==
  /esprima/4.0.1:
    dev: false
    engines:
      node: '>=4'
    hasBin: true
    resolution:
      integrity: sha512-eGuFFw7Upda+g4p+QHvnW0RyTX/SVeJBDM/gCtMARO0cLuT2HcEKnTPvhjV6aGeqrCB/sbNop0Kszm0jsaWU4A==
  /fs-extra/8.1.0:
    dependencies:
      graceful-fs: 4.2.1
      jsonfile: 4.0.0
      universalify: 0.1.2
    dev: false
    engines:
      node: '>=6 <7 || >=8'
    resolution:
      integrity: sha512-yhlQgA6mnOJUKOsRUFsgJdQCvkKhcz8tlZG5HBQfReYZy46OwLcY+Zia0mtdHsOo9y/hP+CxMN0TU9QxoOtG4g==
  /graceful-fs/4.2.1:
    dev: false
    resolution:
      integrity: sha512-b9usnbDGnD928gJB3LrCmxoibr3VE4U2SMo5PBuBnokWyDADTqDPXg4YpwKF1trpH+UbGp7QLicO3+aWEy0+mw==
  /js-yaml/3.13.1:
    dependencies:
      argparse: 1.0.10
      esprima: 4.0.1
    dev: false
    hasBin: true
    resolution:
      integrity: sha512-YfbcO7jXDdyj0DGxYVSlSeQNHbD7XPWvrVWeVUujrQEoZzWJIRrCPoyk6kL6IAjAG2IolMK4T0hNUe0HOUs5Jw==
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.2.1
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /ramda/0.26.1:
    dev: false
    resolution:
      integrity: sha512-hLWjpy7EnsDBb0p+Z3B7rPi3GDeRG5ZtiI33kJhTt+ORCd38AbAIjB/9zRIUoeTbE/AVX5ZkU7m6bznsvrf8eQ==
  /sprintf-js/1.0.3:
    dev: false
    resolution:
      integrity: sha1-BOaSb2YolTVPPdAVIDYzuFcpfiw=
  /ts-toolbelt/3.6.1:
    dev: false
    peerDependencies:
      typescript: ^3.5.0
    resolution:
      integrity: sha512-W6L8ppD33zVGUYVQEC8Z812ZKoxXckeiQWc0/d2ofKrmTRueTAOjpVl7KIpo0LjiCPiMHPYJPS8mR3o6X0oVKg==
  /tslib/1.10.0:
    dev: false
    resolution:
      integrity: sha512-qOebF53frne81cf0S9B41ByenJ3/IuH8yJKngAX35CmiZySA0khhkovshKK+jGCaMnVomla7gVlIcc3EvKPbTQ==
  /universalify/0.1.2:
    dev: false
    engines:
      node: '>= 4.0.0'
    resolution:
      integrity: sha512-rBJeI5CXAlmy1pV+617WB9J63U6XcazHHF2f2dbJix4XzpUF0RS3Zbj0FGIOCAva5P/d/GBOYaACQ1w+0azUkg==
specifiers:
  '@types/fs-extra': ^8.0.0
  '@types/js-yaml': ^3.12.1
  '@types/node': ^12.7.1
  '@types/ramda': ^0.26.19
  fs-extra: ^8.1.0
  fs-tree-utils: ^0.1.6
  js-yaml: ^3.13.1
  ramda: ^0.26.1
  tslib: ^1.10.0
  unique-temp-path: ^0.1.1
