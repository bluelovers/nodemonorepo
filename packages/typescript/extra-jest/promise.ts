export function snapRejected<X> (promise: Promise<X>) {
  return promise.then(
    value => {
      throw new Error(`Expecting promise to reject but it didn't`)
    },
    reason => {
      expect(reason).toMatchSnapshot()
    }
  )
}
