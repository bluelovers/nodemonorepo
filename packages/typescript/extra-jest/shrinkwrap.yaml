dependencies:
  '@types/fs-extra': 5.0.4
  '@types/js-yaml': 3.11.2
  '@types/node': 10.12.9
  '@types/ramda': 0.26.0
  fs-extra: 7.0.1
  fs-tree-utils: 0.1.6
  js-yaml: 3.12.0
  ramda: 0.26.0
  tslib: 1.9.3
  unique-temp-path: 0.1.1
packages:
  /@types/fs-extra/5.0.4:
    dependencies:
      '@types/node': 10.12.9
    dev: false
    resolution:
      integrity: sha512-DsknoBvD8s+RFfSGjmERJ7ZOP1HI0UZRA3FSI+Zakhrc/Gy26YQsLI+m5V5DHxroHRJqCDLKJp7Hixn8zyaF7g==
  /@types/js-yaml/3.11.2:
    dev: false
    resolution:
      integrity: sha512-JRDtMPEqXrzfuYAdqbxLot1GvAr/QvicIZAnOAigZaj8xVMhuSJTg/xsv9E1TvyL+wujYhRLx9ZsQ0oFOSmwyA==
  /@types/node/10.12.9:
    dev: false
    resolution:
      integrity: sha512-eajkMXG812/w3w4a1OcBlaTwsFPO5F7fJ/amy+tieQxEMWBlbV1JGSjkFM+zkHNf81Cad+dfIRA+IBkvmvdAeA==
  /@types/ramda/0.25.42:
    dev: false
    resolution:
      integrity: sha512-m8IcEWgJZTb47UARqp2d58f+C8yDxpTC6ElU/muO0z1bYhdgf6LHcO2SE9f1Mt3SJnKCh6GtjKEC99hq6vPnDA==
  /@types/ramda/0.26.0:
    dev: false
    resolution:
      integrity: sha512-6FCz2E3SrkgbuV3piRNfjThSRyMbcIxHknnGYPfrg4k55PHnNRR80boVdl1AME/KfDW+FBVmKrKPyRbFo8HjaQ==
  /argparse/1.0.10:
    dependencies:
      sprintf-js: 1.0.3
    dev: false
    resolution:
      integrity: sha512-o5Roy6tNG4SL/FOkCAN6RzjiakZS25RLYFrcMttJqbdd8BWrnA+fGz57iN5Pb06pvBGvl5gQ0B48dJlslXvoTg==
  /esprima/4.0.1:
    dev: false
    engines:
      node: '>=4'
    hasBin: true
    resolution:
      integrity: sha512-eGuFFw7Upda+g4p+QHvnW0RyTX/SVeJBDM/gCtMARO0cLuT2HcEKnTPvhjV6aGeqrCB/sbNop0Kszm0jsaWU4A==
  /fs-extra/7.0.1:
    dependencies:
      graceful-fs: 4.1.15
      jsonfile: 4.0.0
      universalify: 0.1.2
    dev: false
    engines:
      node: '>=6 <7 || >=8'
    resolution:
      integrity: sha512-YJDaCJZEnBmcbw13fvdAM9AwNOJwOzrE4pqMqBq5nFiEqXUqHwlK4B+3pUw6JNvfSPtX05xFHtYy/1ni01eGCw==
  /fs-tree-utils/0.1.6:
    dependencies:
      '@types/fs-extra': 5.0.4
      '@types/node': 10.12.9
      '@types/ramda': 0.25.42
      fs-extra: 7.0.1
      monorepo-shared-assets: 0.0.17
      ramda: 0.25.0
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-qsOeiFlr2zfQXkLziZh5FLxRPHmUm3Cfwj4cTjVVVij1gtHGkS11/6nxGA1l7ndvb2d+iKiQartR0cewodT5zg==
  /graceful-fs/4.1.15:
    dev: false
    resolution:
      integrity: sha512-6uHUhOPEBgQ24HM+r6b/QwWfZq+yiFcipKFrOFiBEnWdy5sdzYoi+pJeQaPI5qOLRFqWmAXUPQNsielzdLoecA==
  /immutable/3.8.2:
    dev: false
    engines:
      node: '>=0.10.0'
    resolution:
      integrity: sha1-wkOZUUVbs5kT2vKBN28VMOEErfM=
  /js-yaml/3.12.0:
    dependencies:
      argparse: 1.0.10
      esprima: 4.0.1
    dev: false
    hasBin: true
    resolution:
      integrity: sha512-PIt2cnwmPfL4hKNwqeiuz4bKfnzHTBv6HyVgjahA6mPLwPDzjDWrplJBMjHUFxku/N3FlmrbyPclad+I+4mJ3A==
  /jsonfile/4.0.0:
    dev: false
    optionalDependencies:
      graceful-fs: 4.1.15
    resolution:
      integrity: sha1-h3Gq4HmbZAdrdmQPygWPnBDjPss=
  /monorepo-shared-assets/0.0.17:
    dependencies:
      '@types/node': 10.12.9
      '@types/ramda': 0.25.42
      immutable: 3.8.2
      ramda: 0.25.0
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
    resolution:
      integrity: sha512-A0Y2mZ06vwBR0fyZdZZPn5MJcaIcxfs7BGGv9SU82c6e6k6UwnEQWmVx4LkiVvlnwgKI7+MVIFwMwTZhRGM+eQ==
  /ramda/0.25.0:
    dev: false
    resolution:
      integrity: sha512-GXpfrYVPwx3K7RQ6aYT8KPS8XViSXUVJT1ONhoKPE9VAleW42YE+U+8VEyGWt41EnEQW7gwecYJriTI0pKoecQ==
  /ramda/0.26.0:
    dev: false
    resolution:
      integrity: sha512-maK1XqpgsOo5DwjhROjqDvpm1vkphLQbpleVv+b3t5Y9uOQ0t8hTHT582+mDs7RLrex1kd4lWYizNXWLVjsq9w==
  /sprintf-js/1.0.3:
    dev: false
    resolution:
      integrity: sha1-BOaSb2YolTVPPdAVIDYzuFcpfiw=
  /tslib/1.9.3:
    dev: false
    resolution:
      integrity: sha512-4krF8scpejhaOgqzBEcGM7yDIEfi0/8+8zDRZhNZZ2kjmHJ4hv3zCbQWxoJGz1iw5U0Jl0nma13xzHXcncMavQ==
  /unique-temp-path/0.1.1:
    dependencies:
      '@types/node': 10.12.9
      tslib: 1.9.3
    dev: false
    engines:
      node: '>= 8.9.0'
    hasBin: true
    resolution:
      integrity: sha512-hYlBbvWrpY5sitojnCwIuWH8FNx7CzNyGqG+CbxRDOe3ehqrPmMRj3D2jy3/6C35+VbRBUVy6f2o29y3SyJ/ag==
  /universalify/0.1.2:
    dev: false
    engines:
      node: '>= 4.0.0'
    resolution:
      integrity: sha512-rBJeI5CXAlmy1pV+617WB9J63U6XcazHHF2f2dbJix4XzpUF0RS3Zbj0FGIOCAva5P/d/GBOYaACQ1w+0azUkg==
registry: 'https://registry.npmjs.org/'
shrinkwrapMinorVersion: 9
shrinkwrapVersion: 3
specifiers:
  '@types/fs-extra': ^5.0.4
  '@types/js-yaml': ^3.11.2
  '@types/node': ^10.9.3
  '@types/ramda': ^0.26.0
  fs-extra: ^7.0.0
  fs-tree-utils: ^0.1.6
  js-yaml: ^3.12.0
  ramda: ^0.26.0
  tslib: ^1.9.3
  unique-temp-path: ^0.1.1
