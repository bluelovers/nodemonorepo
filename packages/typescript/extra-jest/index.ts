import * as snap from './lib/snap'
import * as setupTeardown from './lib/setup-teardown'
import * as snapSpawn from './lib/snap-spawn'
export { snap, setupTeardown, snapSpawn }
