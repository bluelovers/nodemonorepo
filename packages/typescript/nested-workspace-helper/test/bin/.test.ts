import {SpawnSyncOptions} from 'child_process'
import * as xjest from 'extra-jest'
import spawn from './lib/spawn'

const snap = x => xjest.snap.unsafe(x)()

const snapSpawn = (argv?: string[], options?: SpawnSyncOptions) =>
  () => snap(spawn(argv, options))

describe('help message', () => {
  describe('from global command', () => {
    it('with --help', snapSpawn(['--help']))
    it('without arguments', snapSpawn())
  })

  describe('from version-management', () => {
    it('as version-management', snapSpawn(['version-management']))
    it('as verman', snapSpawn(['verman']))
  })

  it('from version-management/mismatches', snapSpawn(['version-management', 'mismatches']))
})
