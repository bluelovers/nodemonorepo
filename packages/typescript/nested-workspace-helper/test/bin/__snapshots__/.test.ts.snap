// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help message from global command with --help 1`] = `
"
error: !<tag:yaml.org,2002:js/undefined> ''
signal: null
status: 0
stderr: |+


stdout: |+

  Options:
    --help     Show help                                                 [boolean]
    --version  Show version number                                       [boolean]



"
`;

exports[`help message from global command without arguments 1`] = `
"
error: !<tag:yaml.org,2002:js/undefined> ''
signal: null
status: 1
stderr: |+

  nested-workspace-helper.inrepo.js <cmd> [args]

  Commands:
    nested-workspace-helper.inrepo.js         Manage internal package versions
    version-management <cmd> [args]                              [aliases: verman]

  Options:
    --version  Show version number                                       [boolean]
    --help     Show help                                                 [boolean]

  Not enough non-option arguments: got 0, need at least 1

stdout: |+



"
`;

exports[`help message from version-management as verman 1`] = `
"
error: !<tag:yaml.org,2002:js/undefined> ''
signal: null
status: 1
stderr: |+

  nested-workspace-helper.inrepo.js version-management <cmd> [args]

  Manage internal package versions

  Commands:
    nested-workspace-helper.inrepo.js verman  Manage mismatched internal
    mismatches <directory>                    dependency versions

  Options:
    --version  Show version number                                       [boolean]
    --help     Show help                                                 [boolean]

  Not enough non-option arguments: got 0, need at least 1

stdout: |+



"
`;

exports[`help message from version-management as version-management 1`] = `
"
error: !<tag:yaml.org,2002:js/undefined> ''
signal: null
status: 1
stderr: |+

  nested-workspace-helper.inrepo.js version-management <cmd> [args]

  Manage internal package versions

  Commands:
    nested-workspace-helper.inrepo.js         Manage mismatched internal
    version-management mismatches             dependency versions
    <directory>

  Options:
    --version  Show version number                                       [boolean]
    --help     Show help                                                 [boolean]

  Not enough non-option arguments: got 0, need at least 1

stdout: |+



"
`;

exports[`help message from version-management/mismatches 1`] = `
"
error: !<tag:yaml.org,2002:js/undefined> ''
signal: null
status: 1
stderr: |+

  nested-workspace-helper.inrepo.js version-management mismatches <directory>

  Manage mismatched internal dependency versions

  Positionals:
    directory  Directory of monorepo                           [string] [required]

  Options:
    --version               Show version number                          [boolean]
    --help                  Show help                                    [boolean]
    --update, -u            Automatically update dependency versions to match
                            their source                [boolean] [default: false]
    --checker, -c           Checker to be used
     [string] [choices: \\"ANY\\", \\"EQUAL\\", \\"TILDA_EQUAL\\", \\"CARET_EQUAL\\", \\"EQUAL_MIN\\",
                   \\"TILDA_MIN\\", \\"CARET_MIN\\", \\"EQUAL_OR_ANY\\", \\"TILDA_EQUAL_OR_ANY\\",
                     \\"CARET_EQUAL_OR_ANY\\", \\"EQUAL_MIN_OR_ANY\\", \\"TILDA_MIN_OR_ANY\\",
                         \\"CARET_MIN_OR_ANY\\", \\"ANY_OR_EQUAL\\", \\"ANY_OR_TILDA_EQUAL\\",
                     \\"ANY_OR_CARET_EQUAL\\", \\"ANY_OR_EQUAL_MIN\\", \\"ANY_OR_TILDA_MIN\\",
                               \\"ANY_OR_CARET_MIN\\"] [default: \\"CARET_EQUAL_OR_ANY\\"]
    --jsonOutput, --json    Output json instead of readdable text
                                                        [boolean] [default: false]
    --jsonOutputIndent      JSON output indentation if --jsonOutput is enabled
                                                             [number] [default: 2]
    --noExitStatus          Do not exit with code 1 when there are outdated
                            dependencies                [boolean] [default: false]
    --noPrint, --quiet, -q  Do not print mismatches to stdout
                                                        [boolean] [default: false]

  Not enough non-option arguments: got 0, need at least 1

stdout: |+



"
`;
